### DISCUSSION TITLE
JWT-authentifaction in Wagtail project

### DISCUSSION THREAD

**Discussion by paragliding-nsk** [torchbox/wagtail-grapple#354] [Q&A]
Hello, in the "Wagtail" project with the "wagtail-grapple" library installed, I am doing JWT user authorization using the "django-graphql-jwt" library. The question is, is there another JWT authorization library better suited for "wagtail-grapple"?

**Comment by zerolab** [comment]
The short answer is.. I don't know. So would appreciate hearing your experience with it

**Reply by paragliding-nsk** [reply]
Do I understand correctly that wagtail-grapple does not have functionality for working with GraphQL JWTs, and so far such functionality is not planned? Maybe you are planning to add an abstraction over django-graphql-jwt to wagtail-grapple?
And of course, if I find some new and interesting way to implement the "GraphQL API", then I will definitely share it :-)

**Reply by zerolab** [reply]
At the moment there are no plans for adding an abstraction over `django-graphql-jwt` as:
1. currently no active (or previous) projects demanded it
2. adding another layer and dependency carries its own burden

I would gladly accept a recipe for the docs, though.

**Comment by barkhayot** [comment]
If you're looking for alternatives, you might want to explore:

Graphene-Django: If you're building a GraphQL API in Wagtail using **"wagtail-grapple,"** you can directly use the authentication methods provided by Graphene-Django, which is the underlying GraphQL framework used by **"wagtail-grapple."**

django-rest-framework-simplejwt: If you're working with a mixed API setup that involves both GraphQL and traditional REST endpoints, you could consider using **"django-rest-framework-simplejwt"** for JWT authentication on the REST endpoints.

Custom Solution: Depending on your requirements, you could also implement a custom JWT authentication solution that fits your specific needs.

**Reply by paragliding-nsk** [reply]
Thanks, "django-rest-framework-simplejwt", as seems, is the best solution for REST API, but in my project I have to use GraphQL API.
